lib/Open/index.js

```diff
@@ -71,24 +71,32 @@ module.exports = {
     return directory(source, options);
   },
 
-  s3 : function(client,params, options) {
+  s3: function(client, params, options) {
+    // detect aws-sdk v2 or aws-sdk v3
+    const isAwsSdkV3 = Object.prototype.hasOwnProperty.call(client, 'send');
+
     var source = {
       size: function() {
-        return new Promise(function(resolve,reject) {
-          client.headObject(params, function(err,d) {
-            if (err)
-              reject(err);
-            else
-              resolve(d.ContentLength);
+        if (isAwsSdkV3) {
+          return client.headObject(params).then(function (res) {
```

Comment:
But AWS SDK V3 doesn't have `headObject` function?

Reply:
hmm, yea I have this a try and it doesn't seem to work out. What if you did something like
```js
try {
   return client.headObject(params).promise()
} catch(e) {
   if (e.msg === 'something') {
        return client.headObject(params).then()
   }
}
```

Reply:
@alvin-nt thanks for opening this PR, do you have the bandwidth to add this tweak or should I give it a shot?

Reply:
Hi. Apologies for the long reply. Due to life circumstances, I could not continue this PR yet. If you may, please give a shot to fix my code.

Reply:
Wouldn't it be enough to change `isAwsSdkV3` to this:

                const isAwsSdkV3 = typeof client.send === "function";

Here, at least, it gives `true` with v3 and `false` with v2

Reply:
Hi, Maybe we can do something like:

```typescript
    if (isAwsSdkV3) {
          return client.send(new HeadObjectCommand(params)).then(function (res) {
            return res.ContentLength;
          });
    }
```